cpp/autosar/src/rules/A5-1-3/LambdaExpressionWithoutParameterList.ql

```diff
@@ -21,6 +21,10 @@ where
   not isExcluded(lambda, LambdasPackage::lambdaExpressionWithoutParameterListQuery()) and
   lambdaFunction = lambda.getLambdaFunction() and
   not lambdaFunction.isAffectedByMacro() and
+  // If it has a parameter, then it will have an
+  // explicit parameter list. Therefore, proceed to check only if the lambda
+  // does not have any parameters.
+  not exists (lambdaFunction.getAParameter()) and
```

Comment:
Can you please provide an updated test case? For example, add a macro that expands to a lambda with parameters:
```cpp
#define PARAM_MACRO [](int i) { i; };
```
Then expand that in the test function:
```cpp
PARAM_MACRO // COMPLIANT
```

Reply:
Thanks for the suggestion. I've added this in: [a7dd290](https://github.com/github/codeql-coding-standards/pull/348/commits/a7dd290a650f9997c97763680bbbb8fa3574e0db)